vendor/engines/secure_rooms/app/controllers/secure_rooms/card_readers_controller.rb

```diff
@@ -0,0 +1,82 @@
+module SecureRooms
+
+  class CardReadersController < ApplicationController
+
+    admin_tab :all
+
+    layout "two_column"
+
+    before_action :authenticate_user!
+    before_action :check_acting_as
+    before_action :init_current_facility
+    before_action :init_product
+    before_action :load_card_reader, except: [:index]
+
+    def initialize
+      @active_tab = "secure_rooms"
+      super
+    end
+
+    def index
+      @card_readers = @product.card_readers
+    end
+
+    def new
+    end
+
+    def edit
+    end
+
+    def create
+      if @card_reader.update_attributes(card_reader_params)
+        flash[:notice] = text("create.success")
+        redirect_to facility_secure_room_card_readers_path(current_facility, @product)
+      else
+        flash.now[:error] = text("create.error", message: @user.errors.full_messages.to_sentence)
+        render :new
+      end
+    end
+
+    def update
+      if @card_reader.update_attributes(card_reader_params)
+        flash[:notice] = text("update.success")
+        redirect_to facility_secure_room_card_readers_path(current_facility, @product)
+      else
+        flash.now[:error] = text("update.error", message: @user.errors.full_messages.to_sentence)
```

Comment:
Can we avoid adding errors to the flash message, and add them either through `= f.error_messages` or just relying on simple_form's errors on the form elements themselves?

It's probably fine in this case, but it's caused cookie overflow problems in other models.